pkg/controller/nodes/executor.go

```diff
@@ -811,12 +806,12 @@ func (c *nodeExecutor) isEligibleForRetry(nCtx interfaces.NodeExecutionContext,
 		}
 
 		currentAttempt = (nodeStatus.GetAttempts() + 1) - nodeStatus.GetSystemFailures()
-		maxAttempts = uint32(config.GetConfig().NodeConfig.DefaultMaxAttempts)
-		if nCtx.Node().GetRetryStrategy() != nil && nCtx.Node().GetRetryStrategy().MinAttempts != nil && *nCtx.Node().GetRetryStrategy().MinAttempts != 0 {
-			maxAttempts = uint32(*nCtx.Node().GetRetryStrategy().MinAttempts)
-		}
-		isEligible = currentAttempt < maxAttempts
 	}
+	maxAttempts = uint32(config.GetConfig().NodeConfig.DefaultMaxAttempts)
+	if nCtx.Node().GetRetryStrategy() != nil && nCtx.Node().GetRetryStrategy().MinAttempts != nil && *nCtx.Node().GetRetryStrategy().MinAttempts != 1 {
```

Comment:
```
*nCtx.Node().GetRetryStrategy().MinAttempts != 1 # Changed this 0 -> 1
```

If a user sets `0` retries, `MinAttempts` will be `1`:
https://github.com/fellhorn/flytepropeller/blob/feature/simplify-retries/pkg/compiler/transformers/k8s/utils.go#L24-L25